homeassistant/components/sensor/dht.py

```diff
@@ -128,15 +128,23 @@ def update(self):
         data = self.dht_client.data
 
         if self.type == SENSOR_TEMPERATURE:
-            temperature = data[SENSOR_TEMPERATURE]
+            try:
+                temperature = data[SENSOR_TEMPERATURE]
+            except KeyError:
+                _LOGGER.error("Error reading temperature data")
+                return
```

Comment:
This breaks the situation when the returned data contains humidity data but no temperature data.